plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,389 @@
+                              .add_dim(1, OFF_C / DG - 1, 3)
+                              .add_dim(0, OFF_D)
+                              .add_dim(0, OFF_H)
+                              .add_dim(0, OFF_W);
+  edsl::Tensor offset_w = op::slice(offset)
+                              .add_dim(0, N)
+                              .add_dim(0, DG)
+                              .add_dim(2, OFF_C / DG, 3)
+                              .add_dim(0, OFF_D)
+                              .add_dim(0, OFF_H)
+                              .add_dim(0, OFF_W);
+  offset_d = op::reshape(offset_d, make_tuple<int64_t>({N, DG, F_D, F_H, F_W, OFF_D, OFF_H, OFF_W}));
+  offset_h = op::reshape(offset_h, make_tuple<int64_t>({N, DG, F_D, F_H, F_W, OFF_D, OFF_H, OFF_W}));
+  offset_w = op::reshape(offset_w, make_tuple<int64_t>({N, DG, F_D, F_H, F_W, OFF_D, OFF_H, OFF_W}));
+  offset_d = op::transpose(offset_d, make_tuple<int64_t>({0, 1, 5, 2, 6, 3, 7, 4}));
+  offset_h = op::transpose(offset_h, make_tuple<int64_t>({0, 1, 5, 2, 6, 3, 7, 4}));
+  offset_w = op::transpose(offset_w, make_tuple<int64_t>({0, 1, 5, 2, 6, 3, 7, 4}));
+  offset_d = op::reshape(offset_d, make_tuple<int64_t>({N, DG, 1, OFF_D * F_D, OFF_H * F_H, OFF_W * F_W}));
+  offset_h = op::reshape(offset_h, make_tuple<int64_t>({N, DG, 1, OFF_D * F_D, OFF_H * F_H, OFF_W * F_W}));
+  offset_w = op::reshape(offset_w, make_tuple<int64_t>({N, DG, 1, OFF_D * F_D, OFF_H * F_H, OFF_W * F_W}));
+  offset_d = op::broadcast(offset_d, {N, DG, CI / DG, NEW_D, NEW_H, NEW_W}, {0, 1, 2, 3, 4, 5});
+  offset_h = op::broadcast(offset_h, {N, DG, CI / DG, NEW_D, NEW_H, NEW_W}, {0, 1, 2, 3, 4, 5});
+  offset_w = op::broadcast(offset_w, {N, DG, CI / DG, NEW_D, NEW_H, NEW_W}, {0, 1, 2, 3, 4, 5});
+  offset_d = op::reshape(offset_d, make_tuple<int64_t>({N, CI, NEW_D, NEW_H, NEW_W}));
+  offset_h = op::reshape(offset_h, make_tuple<int64_t>({N, CI, NEW_D, NEW_H, NEW_W}));
+  offset_w = op::reshape(offset_w, make_tuple<int64_t>({N, CI, NEW_D, NEW_H, NEW_W}));
+  // Define depth, hight and width of input.
+  edsl::Tensor index_d_0 = edsl::index({edsl::TensorDim(F_D), edsl::TensorDim(OFF_D)}, static_cast<size_t>(0));
+  edsl::Tensor index_d_1 = edsl::index({edsl::TensorDim(F_D), edsl::TensorDim(OFF_D)}, static_cast<size_t>(1));
+  edsl::Tensor index_d = index_d_0 * dilations[0] + index_d_1 * strides[0] - pad_befores[0];
+  index_d = op::transpose(index_d, make_tuple<int64_t>({1, 0}));
+  index_d = op::reshape(index_d, make_tuple<int64_t>({1, 1, OFF_D * F_D, 1, 1}));
+  index_d = op::broadcast(index_d, {1, 1, NEW_D, NEW_H, NEW_W}, {0, 1, 2, 3, 4});
+  edsl::Tensor index_h_0 = edsl::index({edsl::TensorDim(F_H), edsl::TensorDim(OFF_H)}, static_cast<size_t>(0));
+  edsl::Tensor index_h_1 = edsl::index({edsl::TensorDim(F_H), edsl::TensorDim(OFF_H)}, static_cast<size_t>(1));
+  edsl::Tensor index_h = index_h_0 * dilations[1] + index_h_1 * strides[1] - pad_befores[1];
+  index_h = op::transpose(index_h, make_tuple<int64_t>({1, 0}));
```

Comment:
If you exchange `index_h_0` and `index_h_1`, will the transpose be removed? like
`edsl::Tensor index_h = index_h_1 * dilations[1] + index_h_0 * strides[1] - pad_befores[1];`

Reply:
It will get the wrong value

Reply:
Ok, I think the following is correct.. have to exchange F_H and OFF_H as well, please check..
```
  edsl::Tensor index_h_0 = edsl::index({edsl::TensorDim(F_H), edsl::TensorDim(OFF_H)}, static_cast<size_t>(0));
  edsl::Tensor index_h_1 = edsl::index({edsl::TensorDim(F_H), edsl::TensorDim(OFF_H)}, static_cast<size_t>(1));
  edsl::Tensor index_h = index_h_0 * dilations[1] + index_h_1 * strides[1] - pad_befores[1];
  index_h = op::transpose(index_h, make_tuple<int64_t>({1, 0}));
```

```
  edsl::Tensor index_h_0 = edsl::index({edsl::TensorDim(OFF_H), edsl::TensorDim(F_H)}, static_cast<size_t>(0));
  edsl::Tensor index_h_1 = edsl::index({edsl::TensorDim(OFF_H), edsl::TensorDim(F_H)}, static_cast<size_t>(1));
  edsl::Tensor index_h = index_h_0 *strides[1]  + index_h_1 *dilations[1]  - pad_befores[1];
```

Reply:
I just test it, and it is correct. thanks, i will fix it.